webpack.config.js

```diff
@@ -65,8 +80,53 @@ const config = {
 				use: 'babel-loader',
 			},
 			{
+				test: /block\.s?css$/,
+				use: BlocksCSSPlugin.extract( {
+					use: [
+						{ loader: 'raw-loader' },
+						{ loader: 'postcss-loader' },
+						{
+							loader: 'sass-loader',
+							query: {
+								includePaths: [ 'editor/assets/stylesheets' ],
+								data: '@import "variables"; @import "mixins"; @import "animations";@import "z-index";',
+								outputStyle: 'production' === process.env.NODE_ENV ?
+									'compressed' : 'nested',
+							},
+						},
+					],
+				} ),
+			},
+			{
 				test: /\.s?css$/,
-				use: ExtractTextPlugin.extract( {
+				include: [
+					/blocks/,
+				],
+				exclude: [
+					/block\.s?css$/,
+				],
+				use: EditBlocksCSSPlugin.extract( {
+					use: [
+						{ loader: 'raw-loader' },
+						{ loader: 'postcss-loader' },
+						{
+							loader: 'sass-loader',
+							query: {
+								includePaths: [ 'editor/assets/stylesheets' ],
+								data: '@import "variables"; @import "mixins"; @import "animations";@import "z-index";',
+								outputStyle: 'production' === process.env.NODE_ENV ?
+									'compressed' : 'nested',
+							},
+						},
+					],
+				} ),
+			},
+			{
+				test: /\.s?css$/,
+				exclude: [
+					/blocks/,
+				],
+				use: MainCSSExtractTextPlugin.extract( {
```

Comment:
There is a lot of duplication here.  Can we split this logic out into a function?